nchelpers/__init__.py

```diff
@@ -331,56 +406,117 @@ def climo_periods(self):
+            # "MIP table / table_id" in
+            # https://pcic.uvic.ca/confluence/display/CSG/PCIC+metadata+standard+for+downscaled+data+and+hydrology+modelling+data
+            # Specifically, we do not consult the value of the attribute table_id because it is too limited for our
+            # needs. Instead we map the file's time resolution to a value.
+            components.update(
+                time_range=self.time_range_formatted,
+                mip_table = tres_to_mip_table and tres_to_mip_table.get(self.time_resolution, None)
+            )
+
+        if self.is_unprocessed_gcm_output:
+            components.update(
+                model=self.metadata.model,
+                experiment=self.metadata.emissions,
+            )
+        elif self.is_downscaling_output:
+            components.update(
+                downscaling_method=self.downscaling_method_id,
+                model=self.driving_model_id,
+                experiment=_join_comma_separated_list(self.driving_experiment_id),
+                geo_info=getattr(self, 'domain', None)
+            )
+        elif self.is_hydromodel_dgcm_output:
+            components.update(
+                hydromodel_method=_join_comma_separated_list(self.hydromodel_method_id),
+                model=self.driving_model_id,
+                experiment=_join_comma_separated_list(self.driving_experiment_id),
+                geo_info=getattr(self, 'domain', None)
+            )
+        elif self.is_hydromodel_iobs_output:
+            raise NotImplementedError
+            # TODO: props for observational data info
+            # components.update(
+            #     hydromodel_method=_join_comma_separated_list(self.hydromodel_method_id),
+            #     geo_info=getattr(self, 'domain', None)
+            # )
+
+        # Override with supplied args
+        components.update(**override)
+
+        return components
+
+    @property
     def cmor_filename(self):
         """A CMOR standard filename for this file, based on its metadata contents"""
-        return cmor_filename(
-            variable='+'.join(self.dependent_varnames),
-            mip_table=self.time_resolution,
-            model=self.metadata.model,
-            experiment=self.metadata.emissions,
-            ensemble_member=self.metadata.run,
-            time_range=self.time_range_formatted,
-        )
+        return cmor_type_filename(extension='.nc', **self._cmor_type_filename_components())
 
     @property
     def unique_id(self):
         """A unique id for this file, based on its CMOR filename"""
-        unique_id = self.cmor_filename
+        unique_id = cmor_type_filename(**self._cmor_type_filename_components())
 
         dim_axes = set(self.dim_axes_from_names().keys())
         if not (dim_axes <= {'X', 'Y', 'Z', 'T'}):
             unique_id += "_dim" + ''.join(sorted(dim_axes))
 
-        return unique_id.replace('+', '-')
+        return unique_id.replace('+', '-')  # In original code, but why?
```

Comment:
IIRC this was originally a bad hack to avoid ncwms not interpreting the + as a separator for query parameters. Not sure why the + wasn't just urlencoded...